multimodal/src/autogluon/multimodal/models/ner_text.py

```diff
@@ -0,0 +1,206 @@
+import logging
+from typing import List, Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+from transformers import AutoTokenizer
+from transformers import logging as hf_logging
+
+from ..constants import (
+    AUTOMM,
+    COLUMN,
+    COLUMN_FEATURES,
+    FEATURES,
+    LABEL,
+    LOGITS,
+    MASKS,
+    NER_ANNOTATION,
+    TEXT_SEGMENT_IDS,
+    TEXT_TOKEN_IDS,
+    TEXT_VALID_LENGTH,
+    TOKEN_WORD_MAPPING,
+    WORD_OFFSETS,
+)
+from .huggingface_text import HFAutoModelForTextPrediction
+from .utils import DummyLayer, assign_layer_ids, get_column_features, get_hf_config_and_model, init_weights
+
+hf_logging.set_verbosity_error()
+
+logger = logging.getLogger(AUTOMM)
+
+
+class HFAutoModelForNER(HFAutoModelForTextPrediction):
+    """
+    Named entity recognition with huggingface backbones. Inherit from HFAutoModelForTextPrediction.
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        checkpoint_name: str = "microsoft/deberta-v3-base",
+        num_classes: Optional[int] = 0,
+        pooling_mode: Optional[str] = "cls",
+        gradient_checkpointing: Optional[bool] = False,
+        pretrained: Optional[bool] = True,
+    ):
+        """
+        Load a pretrained huggingface text transformer backbone.
+        Parameters
+        ----------
+        prefix
+            The model prefix.
+        checkpoint_name
+            Name of the checkpoint. We support loading checkpoint from
+            Huggingface Models list: https://huggingface.co/models
+            For example, you may use
+                English backbones:
+                    - 'bert-base-cased'
+        num_classes
+            The number of classes. 1 for a regression task.
+        pooling_mode
+            The pooling mode to be used, it is not used in the NER task.
+        gradient_checkpointing
+            Whether to enable gradient checkpointing
+        pretrained
+            Whether using the pretrained weights. If pretrained=True, download the pretrained model.
+        """
+        super().__init__(prefix, checkpoint_name, num_classes, pooling_mode, gradient_checkpointing, pretrained)
+        logger.debug(f"initializing {checkpoint_name}")
+
+        if self.config.model_type in {"gpt2", "roberta"}:
+            # Refer to this PR: https://github.com/huggingface/transformers/pull/12116
+            self.tokenizer = AutoTokenizer.from_pretrained(checkpoint_name, add_prefix_space=True)
+        else:
+            self.tokenizer = AutoTokenizer.from_pretrained(checkpoint_name)
+
+    @property
+    def text_token_word_mapping_key(self):
+        return f"{self.prefix}_{TOKEN_WORD_MAPPING}"
+
+    @property
+    def text_word_offsets_key(self):
+        return f"{self.prefix}_{WORD_OFFSETS}"
+
+    def forward(
```

Comment:
In a later pr we may refactor the forward code in HFAutoModelForTextPrediction a bit to reduce duplicate codes.